compiler/rustc_hir_analysis/src/check/check.rs

```diff
@@ -364,6 +393,97 @@ fn check_opaque_meets_bounds<'tcx>(
     }
 }
 
+fn best_definition_site_of_opaque<'tcx>(
```

Comment:
This is kinda annoyingly duplicated w/ the TAIT visitor, but also I could not find a way to genericize that without making it a mess, since it has a lot of responsibilities.